app/src/main/java/com/sschuraytz/charcoaldrawing/SaveDrawing.java

```diff
@@ -0,0 +1,150 @@
+                outputDir.mkdirs(); //creates if doesn't exist
+                File newImageFile = new File(path + File.separator + displayName + ".png");
+                FileOutputStream outputStream = new FileOutputStream(newImageFile);
+                bitmap.compress(format, 100, outputStream);
+                outputStream.flush();
+                outputStream.close();
+
+                //Android only runs full media scan on reboot, to add drawing to media library immediately:
+                //pass newly created image file to media scanner service to be added to the phone gallery
+                MediaScannerConnection.scanFile(baseContext, new String[]{newImageFile.toString()},
+                        new String[]{newImageFile.getName()},
+                        (path1, uri) -> Toast.makeText(baseContext, newImageFile.getName() + "was saved to gallery", Toast.LENGTH_SHORT).show());
+            }
+        } catch (IOException ioe) {
+            ioe.getMessage();
+        }
+    }
+
+    public void checkExternalStoragePermissions() {
+        if (ContextCompat.checkSelfPermission(
+                baseActivity, Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(
+                    baseActivity,
+                    new String[] { Manifest.permission.WRITE_EXTERNAL_STORAGE },
+                    1
+            );
+
+        }
+    }
+
+
+/*
+        long id = ContentUris.parseId(url);
+        Bitmap miniThumb = MediaStore.Images.Thumbnails.getThumbnail(contentResolver, id, MediaStore.Images.Thumbnails.MINI_KIND, null);
+        storeThumbnail(contentResolver, miniThumb, id, 50F, 50F, MediaStore.Images.Thumbnails.MICRO_KIND);
+ */
+
+/*    private Bitmap storeThumbnail(
+            ContentResolver contentResolver, Bitmap bitmap, long id,
+            float width, float height, int kind) {
+        Matrix matrix = new Matrix();
+
+        float scaleX = width / bitmap.getWidth();
+        float scaleY = height / bitmap.getHeight();
+
+        matrix.setScale(scaleX, scaleY);
+
+        Bitmap thumb = Bitmap.createBitmap(bitmap, 0, 0,
+                bitmap.getWidth(),
+                bitmap.getHeight(), matrix, true);
+
+        ContentValues values = new ContentValues(4);
+        values.put(MediaStore.Images.Thumbnails.KIND, kind);
+        values.put(MediaStore.Images.Thumbnails.IMAGE_ID, id);
+        values.put(MediaStore.Images.Thumbnails.WIDTH, thumb.getWidth());
+        values.put(MediaStore.Images.Thumbnails.HEIGHT, thumb.getHeight());
+
+        Uri url = contentResolver.insert(MediaStore.Images.Thumbnails.EXTERNAL_CONTENT_URI, values);
+
+        try {
+            OutputStream thumbOut = contentResolver.openOutputStream(url);
+            thumb.compress(Bitmap.CompressFormat.JPEG, 100, thumbOut);
+            thumbOut.close();
+            return thumb;
+        } catch (FileNotFoundException ex) {
```

Comment:
A FileNotFoundException is an IOException, so having two catch clauses is redundant